### PR TITLE
Modernized CMake

### PR DESCRIPTION
**Pull request by mmha**
* Add QtCreator-specific files to .gitignore
* Raised required CMake version to 3.4.3
* Create and install and interface target for SYCL-BLAS
* Add options to disable building the tests/benchmarks/doc
    * Documentation switch without function for now (-> follow-up patch coming)
* Package SYCL-BLAS
    * Users can find_package(SYCL-BLAS) and `target_link_libraries(app PRIVATE codeplay::sycl-blas)` now
* Properly `find_package` CLBlast, BLAS and GTest
* Updated FindComputeCpp to newest version
* Fixups to make SYCL-BLAS compile with newest ComputeCpp
* Also introduced inline variables to avoid ODR violations for now (C++17 feature)
* Removed template parameters from `default_allocator`

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,43 +1,59 @@
...
-  set(BLAS_INCLUDE_DIRS "${OPENBLAS_ROOT}/include/")
-else()
-  message(STATUS "Using Cmake FindBLAS to locate a BLAS library for unit tests")
-  set(BLA_STATIC on)
-  # If we want to use a specific BLAS vendor, we could set it here:
-  # by calling: set(BLA_VENDOR OpenBLAS) 
-  find_package(BLAS REQUIRED) # We need BLAS for the tests - require it
-  message(STATUS "Found BLAS library at: ${BLAS_LIBRARIES}")
+if(BUILD_BENCHMARKS)
+    add_subdirectory(bench)
```
Comment (mehdi-goli) [inline CMakeLists.txt]: this should be benchmark

**Comment by mmha** [comment]
There are a few rough edges here and there, and I need to reenable CI, but it should compile and pass all the tests now.

So feel free to start reviewing. I can try to split off a  few smaller PRs.

**Comment by mehdi-goli** [comment]
Splitting up the generated files. Syclblas.h should have all the .h files. This is used by third party. sycl_blas.hpp should have all the .hpp files used by third party. include travis.

File: tools/CMakeLists.txt [hunk 2]
```diff
@@ -22,8 +22,5 @@
 # *  @filename CMakeLists.txt
 # *
 # **************************************************************************/
-add_subdirectory(blas1)
-add_subdirectory(blas2)
-add_subdirectory(blas3)
-
 
+# add_subdirectory(auto_tuner)
```
Comment (mmha) [inline tools/CMakeLists.txt]: TODO reenable this.

File: Dockerfile [hunk 3]
```diff
@@ -1,67 +0,0 @@
-FROM ubuntu:xenial
```
Comment (AdamHarries) [inline Dockerfile]: Deleting this file breaks building on travis. Why has it been deleted?

**Comment by Rbiessy** [comment]
I'm afraid these changes are outdated, should we close the MR?

**Comment by mehdi-goli** [comment]
> I'm afraid these changes are outdated, should we close the MR?

I think so

**Pull request closed without merging**